src/main/java/com/amplitude/HttpCall.java

```diff
@@ -89,6 +90,12 @@ protected Response makeRequest(List<Event> events) throws AmplitudeInvalidAPIKey
       timesOutResponse.put("status", Status.TIMEOUT);
       timesOutResponse.put("code", 408);
       responseBody = Response.populateResponse(timesOutResponse);
+    } catch (JSONException e) {
+      // Some error responses from load balancers and reverse proxies may have
+      // response bodies that are not JSON (e.g. HTML, XML).
+      JSONObject decodeFailureResponse = new JSONObject();
+      decodeFailureResponse.put("code", responseCode);
+      responseBody = Response.populateResponse(decodeFailureResponse);
```

Comment:
Question: we only put the responseCode in responseBody here. Should we try to decode it since we might know it's type will be HTML or XML?

Reply:
Response code is accessed from the http response (protocol) rather than from the actual response body. We don't necessarily care what is in the response body in if it's not json, just the code.

Reply:
The JSON object here wrapping the code is used to conform to what the application protocol expects, which is a `JSONObject` with the response code included.